core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1211,6 +1211,7 @@ public Builder removeSnapshots(Collection<Long> idsToRemove) {
         if (idsToRemove.contains(snapshotId)) {
           snapshotsById.remove(snapshotId);
           changes.add(new MetadataUpdate.RemoveSnapshot(snapshotId));
+          removeStatistics(snapshotId);
```

Comment:
Can you explain why this PR is changing the behavior of `TableMetadata`?

Reply:
expire snapshots used to remove the expired snapshot entires from `TableMetadata` (which is changing contents of `TableMetadata`).
So, now that those snapshots are expired. We need to clean up the statistics files entires present for those expired snapshots. Hence, modifying the  `TableMetadata`